go/cmd/dolt/commands/sqlserver/serverconfig.go

```diff
@@ -392,6 +393,9 @@ func ValidateConfig(config ServerConfig) error {
 
 // ConnectionString returns a Data Source Name (DSN) to be used by go clients for connecting to a running server.
 func ConnectionString(config ServerConfig) string {
+	if runtime.GOOS != "windows" && config.Host() == "localhost" {
```

Comment:
Is `localhost` enough here? Should we also include `127.0.0.1`? I'm guessing you've probably already checked MySQL's behavior.

Reply:
We probably also want an option on `dolt sql-server` to disable unix socket support.

Reply:
If nothing is given along with running `dolt sql-server` it will use default host, https://github.com/dolthub/dolt/pull/3796/files#diff-e137b5485cf44698ce998c1e76e36fb0a608954e38f084323c87b04db6ba662fR39 , which is defined as `localhost` so it'll be always localhost in case of not defining host and port when running a server.

I agree we should add option for disable unix socket support, but this would require host and port number which it can fallback to default host `127.0.0.1` and port `3306` if none given I think.